### PR TITLE
Add 3 tier model mapping support

### PR DESCRIPTION
**Pull request by hikinggrass**
Add module level evse and connector mappings

Add 3 tier model mappings to errors

### CODE REVIEW

File: schemas/config.yaml [hunk 1]
```diff
@@ -147,6 +147,36 @@ properties:
             default: {}
             # don't allow arbitrary additional properties
             additionalProperties: false
+          evse:
```
Comment (Pietfried) [inline schemas/config.yaml:150]: nit / personal preference: `evse_id`

File: schemas/config.yaml [hunk 2]
```diff
@@ -147,6 +147,36 @@ properties:
...
+            description: evse this module id maps to
+            type: integer
+          connector:
```
Comment (Pietfried) [inline schemas/config.yaml:153]: nit / personal preference: `connector_id`

File: include/utils/types.hpp [hunk 3]
```diff
@@ -84,6 +84,22 @@ struct TelemetryConfig {
     int id;
 };
 
+struct Mapping {
```
Comment (Pietfried) [inline include/utils/types.hpp:88]: I think a brief description for the struct would help here, same for `ModuleTierMappings`

**Review by Pietfried — APPROVED** [review]
Added a few suggestions

**Comment by Pietfried** [comment]
We should also add this new functionality to the documentation, maybe here: https://everest.github.io/nightly/general/05_existing_modules.html#everest-module-configurations 
@krealyt

File: schemas/config.yaml (hunk 1 above)
Comment (hikinggrass) [reply schemas/config.yaml:150]: my thinking here was that this gets a bit verbose, but I don't mind one way or the other. But if we change it we should also change it in the impl/provides mapping

File: include/utils/types.hpp (hunk 3 above)
Comment (hikinggrass) [reply include/utils/types.hpp:88]: I'll add some more documentation for everything that was added in this pr :+1:

**Pull request merged**